### PR TITLE
[SYCL] Fix accessor_iterator operator-

### PR DESCRIPTION
**Pull request by KornevNikita**

### CODE REVIEW

File: sycl/include/sycl/detail/accessor_iterator.hpp [hunk 1]
```diff
@@ -105,10 +105,10 @@ template <typename DataT, int Dimensions> class accessor_iterator {
     return *this;
   }
 
-  friend accessor_iterator operator-(accessor_iterator &Lhs,
+  friend accessor_iterator operator-(const accessor_iterator &Lhs,
                                      difference_type N) {
-    Lhs -= N;
-    return Lhs;
+    accessor_iterator temp = Lhs;
+    return temp -= N;
```
Comment (AlexeySachkov) [inline sycl/include/sycl/detail/accessor_iterator.hpp:111]:
```suggestion
    auto Temp = Lhs;
    return Temp -= N;
```

File: sycl/unittests/accessor/AccessorIterator.cpp [hunk 2]
```diff
@@ -277,6 +277,24 @@ TEST_F(AccessorIteratorTest, LegacyRandomAccessIteratorRequirementsExtra) {
...
+    // return temp -= n;
+    auto It1 = accessor.end();
+    auto It2 = accessor.end();
+
+    It2 -= 3;
+    ASSERT_EQ(It1 - 3, It2);
+    ASSERT_EQ(It1, accessor.end());
+  }
+  {
+    const auto It1 = accessor.end();
```
Comment (AlexeySachkov) [inline sycl/unittests/accessor/AccessorIterator.cpp:292]: Let's leave a comment that here we test `it - n` where `it` is const. Perhaps we could even merge this into the section about which you added, just as `const auto It3 = ...; ASSERT_EQ(It3 - 3, It2);`

Comment (KornevNikita) [reply sycl/unittests/accessor/AccessorIterator.cpp]: This one and the comment above - done 1aa5d7f

**Review by AlexeySachkov — APPROVED** [review]

**Comment by steffenlarsen** [comment]
AtomicRef failures on CUDA are addressed in https://github.com/intel/llvm/commit/e99ead8cf00f9bbca3cd78f09c02a62d638e0b15.

**Pull request merged**